src/query/parser/promql/parse.go

```diff
@@ -160,6 +160,40 @@ func (p *parseState) addLazyOffsetTransform(offset time.Duration) error {
 	return nil
 }
 
+func (p *parseState) addLazySubqueryTransform(sbq *pql.SubqueryExpr) error {
+	tt := func(t time.Time) time.Time {
+		return t.Add(sbq.Offset)
+	}
+
+	mt := func(meta block.Metadata) block.Metadata {
+		meta.Bounds.Start = meta.Bounds.Start.Add(sbq.Offset)
+		meta.Bounds.Duration = meta.Bounds.Duration + sbq.Range
```

Comment:
Hm, this seems like it extends the range that it covers rather than just moving the datapoints back to the original range.

Is this desired?  I would've though that the fetch ends up being for the same time window width (i.e. duration)?